components/sources/cu_hesai/src/lib.rs

```diff
@@ -121,7 +121,12 @@ impl<'cl> CuSrcTask<'cl> for Xt32 {
                         payload.push(LidarPayload::new_uom(t, x, y, z, r, None));
                     });
                 }
-                new_msg.metadata.tov = payload.tov[0].into(); // take the oldest timestamp
+
+                let tov_range = CuTimeRange {
+                    start: payload.tov[0],
```

Comment:
Since this is a soa, can we build a CuTimeRange from a slice?

So impl From or TryFrom for CuTimeRange and get `CuTimeRange::from(&[T])`

Syntactic sugar but still useful

Reply:
Added it. It is not used yet but it is a nice thing to have.